src/main/java/org/jboss/staxmapper/XMLMapperImpl.java

```diff
@@ -100,10 +108,24 @@ public void deparseDocument(XMLContentWriter contentWriter, XMLStreamWriter stre
         contentWriter.writeContent(new FormattingXMLStreamWriter(streamWriter));
     }
 
-    @SuppressWarnings({ "unchecked" })
+    private XMLElementReader<?> getParser(final QName name) {
+        rootElements.computeIfAbsent(name, qName -> {
+            XMLElementReader<?> reader = rootElementsProvider.get(name).get(); //lazy create of parser
+            return new WeakReference<>(reader);
```

Comment:
I don't understand what problem it solves though (no JIRA?).  Once parsing is done, the whole thing should be discarded, right?